ci/svd2rust-regress/src/svd_test.rs

```diff
@@ -386,3 +446,30 @@ fn visit_dirs(dir: &Path, cb: &mut dyn FnMut(&fs::DirEntry)) -> std::io::Result<
     }
     Ok(())
 }
+
+pub struct WorkspaceTomlGuard {
+    pub unmodified_toml: Vec<u8>,
+    pub manifest_path: PathBuf,
+}
+
+impl WorkspaceTomlGuard {
+    pub fn new() -> Self {
+        // XXX: Workaround for
+        // https://github.com/rust-lang/cargo/issues/6009#issuecomment-1925445245
+        // Calling cargo init will add the chip test directory to the workspace members, regardless
+        // of the exclude list. The unmodified manifest path is stored here and written
+        // back after cargo init.
+        let manifest_path = crate::get_cargo_workspace().join("Cargo.toml");
+        let unmodified_toml = fs::read(&manifest_path).unwrap();
+        Self {
+            unmodified_toml,
+            manifest_path,
+        }
+    }
```

Comment:
im not sure this is needed anymore, due to us adding `[workspace]` to the manifest